src/machine.rs

```diff
@@ -478,7 +478,24 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
                 (None, Tag::Untagged)
             };
         let race_alloc = if let Some(data_race) = &memory_extra.data_race {
-            Some(data_race::AllocExtra::new_allocation(&data_race, alloc.size))
+            match kind {
+                // V-Table generation is lazy and so racy, so do not track races.
+                // Also V-Tables are read only so no data races can be detected.
```

Comment:
```suggestion
                // Also V-Tables are read only so no data races can occur.
```

Reply:
Also please explain that vtables are different from other global memory because we use interpreter writes to initialize them -- it's those writes that cause the race detector to fire.